homeassistant/components/sensor/openweathermap.py

```diff
@@ -164,20 +178,28 @@ def update(self):
             else:
                 self._state = 'not snowing'
                 self._unit_of_measurement = ''
```

Comment:
I know it's not part of your current commit, but your class is not recovering from setting `unit_of_measurement` to '' here. As in: if it starts snowing/raining, the unit does not exist anymore?